### ISSUE TITLE
pyright 1.1.391 checks against python std libs unexpectedly and broken

### ISSUE THREAD

**Issue by Juniormin123** [microsoft/pyright#9606]
**Describe the bug**
`pyright` 1.1.391 checks against python std libs unexpectedly and broken.

**Code or Screenshots**

Environment:

```
Linux ... 6.12.4-1-MANJARO #1 SMP PREEMPT_DYNAMIC Mon, 09 Dec 2024 11:58:37 +0000 x86_64 GNU/Linux
Python 3.11.11 (main, Dec 11 2024, 16:28:39) [GCC 11.2.0] on linux
```

Minimal example:

1. New conda env

```bash
conda create -n pyright-test python=3.11
conda activate pyright-test
pip install pyright
```

2. Make the following file
```
(pyright-test) ~/temp/pyright_issue via  v3.11.11 via  pyright-test 
➜ tree .      
.
└── test.py
```

```python
# test.py
from typing import Final

a: Final = 1

print(a)
```

3. Run the pyright

```
pyright
```

Output

```
/home/calvin/temp/pyright_issue/test.py
  /home/calvin/temp/pyright_issue/test.py:3:4 - error: Expected class but received "_SpecialForm" (reportGeneralTypeIssues)
  /home/calvin/temp/pyright_issue/test.py:5:1 - error: "print" is not defined (reportUndefinedVariable)
/home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py:4:8 - error: Import "sys" could not be resolved (reportMissingImports)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py:12:16 - error: "type" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py:13:14 - error: "type" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py:14:12 - error: "type" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py:15:20 - error: "type" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py:15:25 - error: "type" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/types.py:16:19 - error: "type" is not defined (reportUndefinedVariable)
...
/home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:8:8 - error: Import "sys" could not be resolved (reportMissingImports)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:138:15 - error: "type" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:155:16 - error: "isinstance" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:256:27 - error: Expected name of TypeVar as first argument (reportGeneralTypeIssues)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:336:4 - error: "hasattr" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:343:52 - error: "dict" is not defined (reportUndefinedVariable)
...
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:3593:17 - error: "getattr" is not defined (reportUndefinedVariable)
  /home/calvin/miniconda3/envs/pyright-test/lib/python3.11/site-packages/typing_extensions.py:3595:28 - error: "type" is not defined (reportUndefinedVariable)
447 errors, 5 warnings, 0 informations 
```

NOTE:

If I downgrade to `1.1.390`, the above issue does not present.

```bash
# in the same conda env
pip install -U pyright==1.1.390

pyright
```

Output from `1.1.390`

```
(pyright-test) ~/temp/pyright_issue via  v3.11.11 via  pyright-test 
➜ pyright                        
WARNING: there is a new pyright version available (v1.1.390 -> v1.1.391).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

0 errors, 0 warnings, 0 informations 
```

**Comment by Juniormin123** [comment]
This same issue cannot be reproduced on a different machine following the above steps.

**Comment by erictraut** [comment]
I'm not able to repro the problem. I get the same output with 1.1.390 and 1.1.391.

Based on the errors you're reporting, it looks like your local copy of pyright 1.1.391 is corrupted. It appears to be unable to locate its bundled typeshed stubs. The fact that you're able to repro this on multiple machines makes my theory less likely, but I don't have any other theories at the moment.

You could try running pyright with "--verbose" to see if there are any additional clues.

I haven't received reports from anyone else encountering this issue, so it's likely that it's specific to you and your system / configuration / environment. Without additional clues or the ability to repro the issue, there's not much I can do to assist. For now, you can stick with 1.1.390 since it's working for you. If my theory about a corrupt copy is correct, it should clear up for you in version 1.1.392. 

If anyone else is seeing similar behaviors with pyright 1.1.391, please report it here in this thread.

**Comment by Joelius300** [comment]
Dear @Juniormin123 
I think I'm running into a similiar issue (see #9685). The only things our environments seem to have in common is that we're on Linux and using conda. I don't get any errors outside of my code, but my code is littered with false positives. Were you able to verify that the installation is broken and replace it with a working one? I haven't figured it out yet.

**Comment by Juniormin123** [comment]
> Were you able to verify that the installation is broken and replace it with a working one? I haven't figured it out yet.

@Joelius300 Hi, I cannot verify the installation's integrity as I am unfamiliar with the `npm` part of the `pyright`. Simply doing a `pip install -U pyright==1.1.391` will not help. I believe if some how we purge the `npm` part, then we can have more insight into this problem.
